src/ipc/ipc-helper.c

```diff
@@ -291,7 +291,7 @@ int ipc_comp_free(struct ipc *ipc, uint32_t comp_id)
 		return -EINVAL;
 	}
 
-	if (!icd->cd->bsource_list.next || !icd->cd->bsource_list.next) {
+	if (!icd->cd->bsource_list.next || !icd->cd->bsink_list.next) {
```

Comment:
First of all please remove the rimage hunk. Secondly maybe we can fix the comment text below too then:
```
	 * have to do it here and eat the resulting memory
	 * leak on error.  Bug-free host drivers won't do
```